fcrepo4/fcrepo4
# Allow single-document ACLs using hash URIs for authorizations.

- Add a PATCH method to accept ACLs document

Resolves: https://jira.duraspace.org/browse/FCREPO-2742

# What does this Pull Request do?
Add a PATCH method to SPARQL-UPDATE ACL resources in the static fcr:acl URI http://resource/path/fcr:acl. 

# How should this be tested?

1. Create RDFSource
```
$ curl -i -XPOST -H "Slug: TestObject" -H "Content-Type: text/n3" --data-binary "<> <info:fedora/test/field> \"Test Object\"" http://localhost:8080/rest/
```
2. Create ACL on the RDFSource and verify status 201 with the location header URI translated to fcr:acl
```
$ curl -i -XPUT http://localhost:8080/rest/TestObject/fcr:acl
HTTP/1.1 201 Created
```
3. Create a authz.sparql
```
PREFIX acl: <http://www.w3.org/ns/auth/acl#>

INSERT DATA {
    <#readAccess> acl:mode acl:write .
}
```
4. Update the ACL authorization
```
curl -i -X PATCH http://localhost:8080/rest/TestObject/fcr:acl \
      -H "Content-Type: application/sparql-update" \
      --data-binary @authz.sparql
```

# Interested parties
Tag @awoods @peichman-umd
